.github/workflows/test_linux_job.yml

```diff
@@ -22,7 +23,22 @@ jobs:
       gpu-arch-version: ""
       script: |
         [[ "${SECRET_NOT_A_SECRET_USED_FOR_TESTING}" == "SECRET_VALUE" ]] || exit 1
+  test-secrets-no-filter-var-forked-repo:
+    # This test secret is not available on forked repo
+    if: github.event.pull_request.head.repo.full_name != github.repository
```

Comment:
Are you checking to see if the PR is being merged into a different repository?

Reply:
Ah no, this is a check to see if the PR comes from a forked repo.  For example, this one comes from https://github.com/huydhn/test-infra/tree/fix-test-linux-job-secret, so `github.event.pull_request.head.repo.full_name` would be `huydhn/test-infra` while `github.repository` would be `pytorch/test-infra`.  On the other hand, if your branch is in `pytorch/test-infra`, both `github.event.pull_request.head.repo.full_name` and `github.repository` will be the same.  Usually, we use `if: github.repository_owner == 'pytorch'` for this check, but it doesn't work in this case (I think this is because linux_job is a reusable workflow)